mqtt_as/mqtt_as.py

```diff
@@ -390,22 +396,17 @@ def close(self):
     async def subscribe(self, topic, qos):
         pkt = bytearray(b"\x82\0\0\0")
         pid = newpid(self.pid)
-        self.sub_pids.add(pid)  # PID's awaiting suback
+        self.pid = pid  # will otherwise result in multiple operations having the same pid
```

Comment:
Simple oversight: pid was not assigned to self.pid so the next generated pid was the same.